### PR TITLE
bpo-45811: Improve error message when source code contains invisible control characters

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue45811

### CODE REVIEW

File: Parser/tokenizer.c [hunk 1]
```diff
@@ -2045,6 +2045,12 @@ tok_get(struct tok_state *tok, const char **p_start, const char **p_end)
         break;
     }
 
+    if (!Py_UNICODE_ISPRINTABLE(c)) {
+        char hex[9];
+        (void)PyOS_snprintf(hex, sizeof(hex), "%04X", c);
+        return syntaxerror(tok, "invalid non-printable character U+%s", hex);
```
Comment (pablogsal) [inline Parser/tokenizer.c:2051]: I am keeping the error simple to keep it consistent with invalid non-printable characters in names

**Comment by pablogsal** [comment]
Example:

```
❯ ./python -c "eval('print\x16(\"Hello\")')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1
    print("Hello")
         ^
SyntaxError: invalid non-printable character U+0016
```

File: Parser/tokenizer.c (hunk 1 above)
Comment (terryjreedy) [reply Parser/tokenizer.c:2051]: Makes sense.

File: lel.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: cp437 -*-
+"┬ó┬ó┬ó┬ó┬ó┬ó" + f(4, x for x in range(1))
```
Comment (terryjreedy) [inline lel.py]: Is adding this to the repository root intended?  What does it do?

Comment (pablogsal) [reply lel.py]: Is not intended, I forgot to push the commit that deleted this file

**Pull request merged**